src/demo.cpp

```diff
@@ -38,44 +38,14 @@ int main(int argc, char** argv)
       // Get the active state and available events
       QStringList active_states = interface.getSM()->activeStateNames();
       const QString& current_state = active_states.at(0);
-      QSet<QString> available_events = map.at(current_state);
+      std::set<std::pair<QString, QString>> available_events = map.at(current_state);
 
       std::stringstream ss;
       ss << "Available events: [ ";
-      for (const QString& event : available_events)
+      for (const auto& pair : available_events)
       {
-        ss << event.toStdString() << " ";
-      }
-      ss << "]";
-
-      // Get user input as to which event to execute
-      bool done = false;
-      while (!done)
-      {
-        std::cout << ss.str() << std::endl;
-
-        // Get the character input
-        auto input = std::cin.get();
-        // Throw away the enter input
-        std::cin.get();
-
-        if (std::isdigit(input))
-        {
-          int idx = static_cast<int>(input) - 48;
-          if (idx < available_events.size())
-          {
-            interface.submitEvent(available_events.toList().at(idx));
-            done = true;
-          }
-          else
-          {
-            std::cout << "Index " << idx << " was not in range [0, " << available_events.size() - 1 << "]" << std::endl;
-          }
-        }
-        else
-        {
-          std::cout << "Input must be numeric" << std::endl;
-        }
```

Comment:
It seems like this change basically removes the functionality of this demo, which I think is still relevant. Is there a reason this was removed?

Reply:
It was commented out when I was working on it and you had made a review comment somewhere to remove unused code. 
I will return it to it's original state.